### DISCUSSION TITLE
Consume keyboard events when editing a `TextEdit`

### DISCUSSION THREAD

**Discussion by margual56** [emilk/egui#3140] [Q&A]
Hello everyone!

### Here's my setup
I have a main element that is always visible and plays/pauses an animation when the spacebar is pressed. 

I also have a `TextEdit::singleline` that is editable.

### The issue is that
When the `TextEdit` is being edited, the keyboard events are not consumed. Thus, when writing an space, the animation is played or paused.

## Question
How can I make the `TextEdit` consume all keyboard events when focused? 

### PS
The actual setup is much more complex, so the solution 
```Rust
let res = ui.add(TextEdit::...); 

if !res.has_focus() && ui.input(|i| i.consume_key(Modifiers::NONE, Key::Space) {
    play_pause();
}
``` 
is not a viable solution and does not work for my use-case.

---

Thank you for reading!

**Comment by SUPERCILEX** [comment]
Did you ever figure this out? I've got the same use case.

**Reply by SUPERCILEX** [reply]
Well throwing this in the if statement before the `TextEdit` works:
```rust
        ui.input_mut(|i| i.events.retain(|e| !matches!(e, Event::Text(_))))
```
